src/shogun/features/DenseFeatures.h

```diff
@@ -60,7 +61,7 @@ class DotFeatures;
  * See comments to find out whether it is supported for that method.
  * See also Features class documentation
  */
-template<class ST> class DenseFeatures: public DotFeatures
+template<class ST> class DenseFeatures: public DotFeatures, public SGContainerIteratorBase<DenseFeatures<ST>, ST>
```

Comment:
mmmm what happens in this case for langs where multi inheritance is not supported (like java)?

Reply:
see swig docs: `Note that Java does not support multiple inheritance so any multiple inheritance in the C++ code is not going to work`

Reply:
I'm not sure actually.... I didn't get any errors/warnings in the tests. The files are exposed in the interfaces via swig. Can hide it from swig?

Reply:
in any case, it says the second class will be ignored, which is fine, the interface shouldn't have access to this functionality right?